src/mocks/mock.ts

```diff
@@ -35,10 +35,10 @@ export class Mock<T> {
 
     /** Extend the current mock object with implementation */
     public extend(object: RecursivePartial<T>): this {
-        Object.keys(object).forEach((key) => {
+        Object.keys(object).forEach((key: keyof T) => {
             if (typeof object[key] === 'function') {
```

Comment:
The object[key] lookup was not passing type typescript 2.5 typecheck.

Type 'string' cannot be used to index type 'Partial<{ [key in keyof T]: T[key] | Partial<any>; }>'.